src/sentry/replays/tasks.py

```diff
@@ -28,22 +31,23 @@ def delete_recording_segments(project_id: int, replay_id: str, **kwargs: dict) -
 
 def delete_replay_recording(project_id: int, replay_id: str) -> None:
     """Delete all recording-segments associated with a Replay."""
-    segments = fetch_segments_metadata(project_id, replay_id, offset=0, limit=10000)
-    for segment in segments:
-        driver = FilestoreBlob() if segment.file_id else StorageBlob()
-        driver.delete(segment)
+    # Delete the segments which are now stored in clickhouse
+    segments_from_metadata = fetch_segments_metadata(project_id, replay_id, offset=0, limit=10000)
+    for segment_metadata in segments_from_metadata:
+        driver = FilestoreBlob() if segment_metadata.file_id else StorageBlob()
+        driver.delete(segment_metadata)
 
-    # delete old rows from SQL too
-    segment_sql_rows = ReplayRecordingSegment.objects.filter(
+    # Delete the ReplayRecordingSegment models that we previously stored using django models
+    segments_from_django_models = ReplayRecordingSegment.objects.filter(
         replay_id=replay_id, project_id=project_id
     ).all()
-    for segment in segment_sql_rows:
-        segment.delete()  # Three queries + one request to the message broker
+    for segment_model in segments_from_django_models:
+        segment_model.delete()  # Three queries + one request to the message broker
 
 
 def archive_replay(project_id: int, replay_id: str) -> None:
     """Archive a Replay instance. The Replay is not deleted."""
-    replay_payload = {
+    replay_payload: dict[str, Any] = {
```

Comment:
https://github.com/getsentry/sentry-kafka-schemas/pull/143

:)

Reply:
weird, i get 
```
src/sentry/replays/tasks.py:50: error: Need type annotation for "replay_payload"  [var-annotated]
```
when removed

Reply:
yeah just saying you could have better types here that are actually enforced across services (incl relay)

Reply:
(deleted my comment, didn't see your link haha). yeah 100%. we should def. work towards that.